Idiomatic rust logging (#1537)
holochain/holochain-rust: logging/Readme.md

# Logging

> Idiomatic Rust logging implementation.

Reviewer: Let's clean up the readme here too.

> This PR is related to [holochain/lib3h#29](https://github.com/holochain/lib3h/issues/29) and implement a more idiomatic logging.

Reviewer 2: ```suggestion
This PR is related to [holochain/lib3h#29](https://github.com/holochain/lib3h/issues/29) and implements a more idiomatic logging.
```

Requirements can be found [here](https://hackmd.io/MP5F3UhSTp2iPk37Cwa-fw).

> It still misses:

Reviewer 2: So these things will get added before merging this PR?

> It still misses:

Reviewer: I think we can remove this from the readme.  It's good enough now.

Reviewer 2: +1

Author: +1

* documentation: will be updated once I get some review from Core
* toml deserialization: will add it once I'm sure this implementation meets the requirements of bot HC Core & Networking.
* configurable output channel: this is more tricky than is sounds because Write is not safe to share between thread, but I have some ideas how to implement at least logging to file.

## Example

To have a generale view of the capability of this crate, you can run this command:

```bash
cargo run --example simple-fastlog-example
```

or

```bash
cargo run --example from-toml-fastlog-example
```

## CLI Support

> This logging factory handles the environnement variable **RUST_LOG** as well so it can be use like this (the logger has to be registered):

Reviewer 2: ```suggestion
This logging factory handles the environnement variable **RUST_LOG** as well so it can be used like this (the logger has to be registered):
```

# Logging

> This logger implementation has been disigned to be fast and delegates all most of the heavy work to a dedicated thread. It offers nice filtering capability that can be combined together along with an optional and easy to use [TOML](https://github.com/toml-lang/toml) configuration.

Reviewer 2: ```suggestion
This logger implementation has been designed to be fast. It delegates most of the heavy work to a dedicated thread. It offers nice filtering capability that can be combined with an optional and easy to use [TOML](https://github.com/toml-lang/toml) configuration.
```
